File: JAVA/01_core_JAVA/Lecture02/pattern6.java
//Code-Diamond
package Lecture02;

import java.util.Scanner;

public class pattern6 {

	public static void main(String[] args) {
		Scanner scn = new Scanner(System.in);
		System.out.print("Enter odd number of rows :");
		int n = scn.nextInt();
		int space = n / 2;
		int star = 1;
		for (int i = 0; i < n; i++) {
			for (int j = 0; j < space; j++) {
				System.out.print(" ");
			}
			for (int k = 0; k < star; k++) {
				System.out.print("*");
			}
			if (i < n / 2) {
				star += 2;
				space--;
				;
			} else {
				star -= 2;
				space++;
			}
			System.out.println();
		}

	}

}

File: C++ New/01_Basic/Lec_02/pattern6.cpp
//Code- diamond
#include<iostream>
using namespace std;
int main(){
    int row;
    cout << "Enter the number of rows :";
    cin >> row;
    int space = row / 2 + 1;
    int star = 1;
    for(int i = 0; i < row; i++){
        for(int i = 0; i < space; i++){
            cout << "  ";
        }
        for(int i = 0; i < star; i++){
            cout << "* ";
        }
        if(i < row / 2){
            space--;
            star += 2 ;
        } else {
            space++;
            star -= 2;
        }
        cout << endl;
    }
    return 0;
}